source/game/kart/KartBurnout.hh

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "game/kart/KartObjectProxy.hh"
+
+namespace Kart {
+
+/// @brief Calculates the duration of burnout and rotation induced
+/// when holding acceleration too long during the race countdown.
+/// @nosubgrouping
+class KartBurnout : protected KartObjectProxy {
+public:
+    KartBurnout();
+    ~KartBurnout();
+
+    void start();
+
+    void calc();
+    void calcRotation();
+    bool calcEnd(u32 duration);
+
+    void activate();
+    void deactivate();
+    [[nodiscard]] bool isActive() const;
```

Comment:
These can be made private. External classes should consult `state()->isBurnout()` to see if a burnout is active.